src/httprpc.cpp

```diff
@@ -240,12 +240,41 @@ static bool HTTPReq_JSONRPC(const std::any& context, HTTPRequest* req)
     return true;
 }
 
+static std::optional<unsigned> StringToOctal(const std::string& str)
+{
+    unsigned ret = 0;
+    for (char c : str) {
+        if (c < '0' || c > '7') return std::nullopt;
+        ret = (ret << 3) | (c - '0');
+    }
+    return ret;
+}
+
+static auto ConvertPermsToOctal(const std::string& str) noexcept -> std::optional<unsigned>
+{
+    // Don't permit setting special bits as they're not relevant to cookie files
+    if (str.length() == 3) return StringToOctal(str);
+    return std::nullopt;
+}
+
 static bool InitRPCAuthentication()
 {
     if (gArgs.GetArg("-rpcpassword", "") == "")
     {
         LogPrintf("Using random cookie authentication.\n");
-        if (!GenerateAuthCookie(&strRPCUserColonPass)) {
+
+        auto cookie_perms{DEFAULT_COOKIE_PERMS};
```

Comment:
I dislike using `auto` both here and the default. It could conceivably not be a type that supports all permission masks.

Reply:
Changed to `fs::perms` here and the default in fcd100cb8e35e7cb00a58a697274a1f2cdb3cc75